tests/scripts/all.sh

```diff
@@ -1179,11 +1179,29 @@ component_test_depends_curves () {
     record_status tests/scripts/curves.pl
 }
 
+component_test_depends_curves_psa () {
```

Comment:
A consequence of this change that I hadn't thought of is that the Travis build now takes too long. Seen in https://api.travis-ci.org/v3/job/725229070/log.txt :
```
The job exceeded the maximum time limit for jobs, and has been terminated.
```
It only got as far as midway through `component_test_depends_pkalgs_psa`.

I don't think all the “depends” tests are useful to have on Travis. They all run on Jenkins as well which can have a heavier load (we use a free service from Travis, but we pay for system that runs Jenkins, so the rate limits on Jenkins is whatever we're prepared to pay, and a few more test runs won't make much difference). The reason we also run them on Travis is that the Jenkins logs are not public (a problem which we'll hopefully solve soon), so when something fails on Jenkins on a pull request by someone who isn't an Arm employee, an Arm employee needs to come along and post the results. It's not that big a deal for these tests since they don't fail very often.

@mpg IIRC you were the main proponent of having the depends tests on Travis. In hindsight, is this still useful? Should we restrict to the previously existing ones, or remove them altogether? Now that Travis [has some non-x86 archictectures](https://docs.travis-ci.com/user/multi-cpu-architectures/), [I'd like to run some tests on at least arm64](https://github.com/ARMmbed/mbedtls/issues/3655), and to keep the load roughly constant, something should go and the depends tests are the most obvious candidate.

Reply:
@jdurkop So I think you should remove the following block from `.travis.yml`:
```
    - name: check compilation guards
      script:
        - tests/scripts/all.sh -k 'test_depends_*' 'build_key_exchanges'
```
with a commit message explaining that these have become too large for the free Travis instance, so they'll now only be run on Jenkins (possibly with a link Gilles's comment above).

Reply:
@mpg Thanks for the suggestion.